### PR TITLE
Feat: Add Catchpoint Exporter

### PR DESCRIPTION
**Pull request by BominRahmani**
#### PR Description
The following adds the [Catchpoint Exporter](https://github.com/grafana/catchpoint-prometheus-exporter) in Alloy. The exporter was created with the intention of supporting the Grafana Catchpoint integration. The exporter is fairly stable, and I wouldn't expect any further changes unless users wanted more metrics.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

**Review by Dasomeone — COMMENTED** [review]
LGTM from integrations point of view, but deferring to the Alloy maintainers for actual approval

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (clayton-cornell) [inline docs/sources/reference/components/prometheus.exporter.catchpoint.md:94]:
```suggestion
[scrape]: ../prometheus.scrape/
[catchpoint_exporter]: https://github.com/grafana/catchpoint-prometheus-exporter
```

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (clayton-cornell) [inline docs/sources/reference/components/prometheus.exporter.catchpoint.md:10]:
```suggestion
The `prometheus.exporter.catchpoint` component embeds the [catchpoint_exporter][] for collecting statistics from a Catchpoint account.
```

What is the Catchpoint exporter embedded into? This sentence needs a little refactoring after this question is answered.

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (clayton-cornell) [inline docs/sources/reference/components/prometheus.exporter.catchpoint.md:18]:
```suggestion
prometheus.exporter.catchpoint "<LABEL>" {
    port              = PORT
    verbosity_logging = <VERBOSITY_LOGGING>
    webhook_path      = <WEBHOOK_PATH>
```

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (clayton-cornell) [inline docs/sources/reference/components/prometheus.exporter.catchpoint.md:82]:
```suggestion
    url = <PROMETHEUS_REMOTE_WRITE_URL>

    basic_auth {
      username = <USERNAME>
      password = <PASSWORD>
```

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (clayton-cornell) [inline docs/sources/reference/components/prometheus.exporter.catchpoint.md:92]:
```suggestion
- _`<PROMETHEUS_REMOTE_WRITE_URL>`_: The URL of the Prometheus remote_write-compatible server to send metrics to.
- _`<USERNAME>`_: The username to use for authentication to the remote_write API.
- _`<PASSWORD>`_: The password to use for authentication to the remote_write API.
```

File: internal/component/prometheus/exporter/catchpoint/catchpoint.go
Comment (wildum) [inline internal/component/prometheus/exporter/catchpoint/catchpoint.go:14]: I don't think that this new component should be GA right away. I think that you should set it to Experimental for now

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (wildum) [inline docs/sources/reference/components/prometheus.exporter.catchpoint.md:31]: these args have default values but are also marked as required which means that the default values can never be used.
They should either be required or have default values

Comment (wildum) [reply docs/sources/reference/components/prometheus.exporter.catchpoint.md:31]: (if you chose to put them optional, you will need to update the args struct:

 `alloy:"verbose_logging,attr"` -> `alloy:"verbose_logging,attr,optional"`

File: internal/converter/internal/staticconvert/internal/build/catchpoint_exporter.go
Comment (wildum) [inline internal/converter/internal/staticconvert/internal/build/catchpoint_exporter.go:1]: converters are used to convert a Grafana agent static mode config to an Alloy config but your converter is not available for the grafana agent static so you don't need to add a converter

File: internal/static/integrations/catchpoint_exporter/catchpoint_exporter.go
Comment (wildum) [inline internal/static/integrations/catchpoint_exporter/catchpoint_exporter.go:24]: just fyi some code from this file is useless like the one above this comment because the yaml part was only relevant for static mode. But since that we did not update the exporters yet to get rid of the legacy part I think it makes sense to do it like this and keep the exporters consistent.

**Review by wildum — COMMENTED** [review]
Thanks for adding this new component, I just have a few little comments, nothing major.

One more request: It would be nice to have a little bit of context in the PR description to explain why do we need the component, what it does, who requested it, is it stable or should we expect important change etc...

File: internal/converter/internal/staticconvert/internal/build/catchpoint_exporter.go
Comment (BominRahmani) [reply internal/converter/internal/staticconvert/internal/build/catchpoint_exporter.go:1]: @wildum Im sorry, I might be misunderstanding here. Would this mean I have to drop all my files related to the static mode or just these files
image

File: internal/static/integrations/catchpoint_exporter/catchpoint_exporter.go
Comment (BominRahmani) [reply internal/static/integrations/catchpoint_exporter/catchpoint_exporter.go:24]: So I should keep it as is?

Comment (wildum) [reply internal/static/integrations/catchpoint_exporter/catchpoint_exporter.go:24]: yes I think this should be kept as it is

File: internal/converter/internal/staticconvert/internal/build/catchpoint_exporter.go
Comment (wildum) [reply internal/converter/internal/staticconvert/internal/build/catchpoint_exporter.go:1]: yeah all the files that are in the converter dir

Comment (BominRahmani) [reply internal/converter/internal/staticconvert/internal/build/catchpoint_exporter.go]: Alright sounds good! I went ahead and removed those files.

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (mattdurham) [inline docs/sources/reference/components/prometheus.exporter.catchpoint.md:29]: "The account to collect metrics for." sounds off here, is there a one to one relation between port and account?

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (mattdurham) [inline docs/sources/reference/components/prometheus.exporter.catchpoint.md:30]: Is this name right? I think it should be `verbose_logging`

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (mattdurham) [inline docs/sources/reference/components/prometheus.exporter.catchpoint.md:31]: This description does not sound correct.

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (BominRahmani) [reply docs/sources/reference/components/prometheus.exporter.catchpoint.md]: This was incorrect, I fixed the descriptions!

File: go.mod
Comment (mattdurham) [inline go.mod:254]: minor nit can we move this up with the other requires?

**Review by mattdurham — APPROVED** [review]
LGTM thank you for your contribution.

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (clayton-cornell) [reply docs/sources/reference/components/prometheus.exporter.catchpoint.md:10]: This question hasn't been answered and the suggestion hasn't been accepted/rejected. 

I feel like "embeds" isn't the right term here.  Maybe its better to state:

```
The `prometheus.exporter.catchpoint` component uses the [catchpoint_exporter][] to collect statistics from a Catchpoint account.
```

Is this still correct?

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (clayton-cornell) [reply docs/sources/reference/components/prometheus.exporter.catchpoint.md:94]: This suggestion is still not accepted/rejected?  If we make the link change as suggested for line 9, then this needs to be added to make sure the link resolves correctly.

**Review by clayton-cornell — APPROVED** [review]
@mattdurham  There's still two open suggestions and one unanswered question.

If we need to get it merged, then don't block on this and I can follow up with another PR.

File: docs/sources/reference/components/prometheus.exporter.catchpoint.md
Comment (BominRahmani) [reply docs/sources/reference/components/prometheus.exporter.catchpoint.md:94]: Sorry, I'm not sure i understand? i added the link as suggested right below the `[scrape]: ../prometheus.scrape/`
image

Comment (clayton-cornell) [reply docs/sources/reference/components/prometheus.exporter.catchpoint.md:93]: As discussed, we can remove this link definition since it's defined in-line on line 9

**Comment by clayton-cornell** [comment]
Docs look OK now. 

One other comment, @wildum suggested this should be flagged experimental was that done? If so we need to add `{{< docs/shared lookup="stability/experimental.md" source="alloy" version="<ALLOY_VERSION>" >}}` to the component doc

**Comment by clayton-cornell** [comment]
@mattdurham All docs questions are answered now. :-)  Good to merge from docs side of things

**Comment by BominRahmani** [comment]
@mattdurham Any chance you can take a last look at this?

**Comment by clayton-cornell** [comment]
@mattdurham Merging - we have a green build pipeline and with your approval and docs approval it should be good to go.

**Comment by mattdurham** [comment]
Thanks @clayton-cornell , apologies @BominRahmani we had a holiday in USA yesterday.

**Pull request merged**
